operator/v1alpha1/types_clusterapi.go

```diff
@@ -0,0 +1,123 @@
+// +kubebuilder:resource:path=clusterapis,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2564
+// +openshift:file-pattern=cvoRunLevel=0000_30,operatorName=cluster-api,operatorOrdering=01
+// +openshift:enable:FeatureGate=ClusterAPIMachineManagement
+
+// ClusterAPI provides configuration for the capi-operator.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.spec) || !has(oldSelf.spec.unmanagedCustomResourceDefinitions) || has(self.spec.unmanagedCustomResourceDefinitions)",message="unmanagedCustomResourceDefinitions cannot be unset once set"
+type ClusterAPI struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +required
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the capi-operator.
+	// +optional
+	Spec *ClusterAPISpec `json:"spec,omitempty"`
+
+	// status defines the observed status of the capi-operator.
+	// +optional
+	Status ClusterAPIStatus `json:"status,omitzero"`
+}
+
+// ClusterAPISpec defines the desired configuration of the capi-operator.
+type ClusterAPISpec struct {
+	// unmanagedCustomResourceDefinitions is a list of ClusterResourceDefinition (CRD)
+	// names that should not be managed by the capi-operator installer
+	// controller. This allows external actors to own specific CRDs while
+	// capi-operator manages others.
+	//
+	// Each CRD name must be a valid DNS-1123 subdomain consisting of lowercase
+	// alphanumeric characters, '-' or '.', and must start and end with an
+	// alphanumeric character, with a maximum length of 253 characters.
+	// Example: "clusters.cluster.x-k8s.io"
+	//
+	// Items cannot be removed from this list once added.
+	//
+	// The maximum number of unmanagedCustomResourceDefinitions is 128.
+	//
+	// +optional
+	// +listType=set
+	// +kubebuilder:validation:MaxItems=128
+	// +kubebuilder:validation:XValidation:rule="oldSelf.all(item, item in self)",message="items cannot be removed from unmanagedCustomResourceDefinitions list"
+	// +kubebuilder:validation:items:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +kubebuilder:validation:items:MaxLength=253
+	UnmanagedCustomResourceDefinitions []string `json:"unmanagedCustomResourceDefinitions,omitempty"`
+}
+
+// ClusterAPIStatus describes the current state of the capi-operator.
+// +kubebuilder:validation:MinProperties=1
```

Comment:
Question: I assume only CAPI-operator would be updating the status of this object. Does the validation here help prevent a specific scenario?

(should be fine to have, just curious why this validation was added)